### ISSUE TITLE
Segmentation fault in command editor menu

### ISSUE THREAD

**Issue by KeinR** [bjorn/tiled#2914]
Steps to reproduce:
* Open the command editor (file -> commands -> edit commands)
* Select the box with the text "press shortcut".
* Select one of the check boxes to the left (either one).

The program will crash with a segmentation fault.

<img width="498" alt="example" src="https://user-images.githubusercontent.com/44657359/95790444-3b07eb80-0c94-11eb-8a97-f27bf2782989.PNG">

_Expected behavior:_ The check box is toggled and the shortcut box is deselected.
_What actually happened:_ The check box is highlighted for a moment, then the program crashes with a segmentation fault.

Debugging with GDB reveals the following stack traces:

If _"Set show output in console"_ was selected:
```
#0  0x00000000004412fb in Tiled::CommandDataModel::setShowOutput(QModelIndex const&, bool) ()
#1  0x000000000044855e in Tiled::CommandsEdit::setShowOutput(int) ()
#2  0x0000000068a948cd in QMetaObject::activate(QObject*, int, int, void**) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Core.dll
#3  0x000000000141c6e0 in QCheckBox::checkStateSet() () from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#4  0x000000000140f56d in QAbstractButton::setIconSize(QSize const&) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#5  0x000000000140f7a5 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#6  0x0000000001375f18 in QWidget::event(QEvent*) () from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#7  0x0000000001337bdc in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#8  0x000000000133fc1a in QApplication::notify(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#9  0x0000000068a694a8 in QCoreApplication::sendSpontaneousEvent(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Core.dll
#10 0x000000000133e018 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPo
inter<QWidget>&, bool, bool) () from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#11 0x000000000138a65d in QDesktopWidget::qt_metacall(QMetaObject::Call, int, void**) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#12 0x000000000138d22b in QDesktopWidget::qt_metacall(QMetaObject::Call, int, void**) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#13 0x0000000001337bdc in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#14 0x000000000133ec23 in QApplication::notify(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#15 0x0000000068a694a8 in QCoreApplication::sendSpontaneousEvent(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Core.dll
#16 0x0000000061976ea4 in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) ()
```

If _"Save map before executing"_ was selected:
```
#0  0x000000000044138b in Tiled::CommandDataModel::setSaveBeforeExecute(QModelIndex const&, bool) ()
#1  0x00000000004484de in Tiled::CommandsEdit::setSaveBeforeExecute(int) ()
#2  0x0000000068a948cd in QMetaObject::activate(QObject*, int, int, void**) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Core.dll
#3  0x000000000126c6e0 in QCheckBox::checkStateSet() () from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#4  0x000000000125f56d in QAbstractButton::setIconSize(QSize const&) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#5  0x000000000125f7a5 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#6  0x00000000011c5f18 in QWidget::event(QEvent*) () from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#7  0x0000000001187bdc in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#8  0x000000000118fc1a in QApplication::notify(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#9  0x0000000068a694a8 in QCoreApplication::sendSpontaneousEvent(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Core.dll
#10 0x000000000118e018 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPo
inter<QWidget>&, bool, bool) () from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#11 0x00000000011da65d in QDesktopWidget::qt_metacall(QMetaObject::Call, int, void**) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#12 0x00000000011dd22b in QDesktopWidget::qt_metacall(QMetaObject::Call, int, void**) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#13 0x0000000001187bdc in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#14 0x000000000118ec23 in QApplication::notify(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Widgets.dll
#15 0x0000000068a694a8 in QCoreApplication::sendSpontaneousEvent(QObject*, QEvent*) ()
   from /cygdrive/c/Program Files/Tiled/Qt5Core.dll
#16 0x0000000061976ea4 in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) ()
```

Tiled version: 1.4.2
OS: Windows 10

**Comment by bjorn** [comment]
Thank you so much for the detailed report! In your screenshot I see you don't actually have a command selected, so all these widgets are supposed to be disabled. I'll have a look at it today.

**Comment by bjorn** [comment]
The fix will be in an eventual 1.4.3, 1.5.0 or the next development snapshot. In the meantime, make sure to select something before using those widgets.

Thanks again for bringing up this issue!
